File: energybalance_pkg/test/f90/test_soilheatflux.f90
!Test generation

PROGRAM test
    USE Soilheatfluxmod
    REAL:: netRadiationEquivalentEvaporation
    REAL:: tau
    REAL:: soilEvaporation
    REAL:: soilHeatFlux
    print *, "--------test_test1_SoilHeatFlux-------"
    tau = 0.9983
    netRadiationEquivalentEvaporation = 638.142
    soilEvaporation = 448.240
    call model_soilheatflux(netRadiationEquivalentEvaporation, tau,  &
            soilEvaporation, soilHeatFlux)
    print *, soilHeatFlux

END PROGRAM

File: energybalance_pkg/test/py/test_soilheatflux.py
'Test generation'

from Soilheatflux import *
from math import *
import numpy as np



def test_test1():
    params= model_soilheatflux(
    tau = 0.9983,
    netRadiationEquivalentEvaporation = 638.142,
    soilEvaporation = 448.240,
     )
    soilHeatFlux_estimated = round(params, 3)
    soilHeatFlux_computed = 188.817
    assert (soilHeatFlux_estimated == soilHeatFlux_computed)